### ISSUE TITLE
VRAWork created by failed batch ingest record not being removed from Images application

### ISSUE THREAD

**Issue by kdid** [nulib/images#347]
### Severity

Is the production site running?
- [x] yes
- [ ] no

Are staff blocked from performing their work?
- [ ] yes
- [x] no

### Descriptive summary
If certain aspects of the batch ingest fail for a record, the VRAWork is not being removed from Images. 

The `multiresimage_batch_worker.rb` is not catching all errors that could lead to a VRA work being left in the application if  step of the ingest fails.

```
    m = Multiresimage.new(pid: pid, vra_xml: ready_xml, from_menu: true)
    m.save

    begin
      # Copy tiff file to tmp directory
      FileUtils.cp(tiff_file, "tmp/#{m.tiff_img_name}")
      m.create_datastreams_and_persist_image_files("tmp/#{m.tiff_img_name}")
    rescue StandardError => e
      m.vraworks.first.delete if m.vraworks.first
      m.delete
      raise "Had a problem saving #{tiff_file}: #{e.message}"
    end
```

Here is an example error where the ingest has failed and the work record is left

```
2016-12-12T20:27:04.420Z 2250 TID-ow6wscu0g WARN: NoMethodError: undefined method `set_attribute' for nil:NilClass

2016-12-12T20:27:04.420Z 2250 TID-ow6wscu0g WARN: /var/www/dil_hydra/releases/20161212165014/app/models/vrawork.rb:67:in `update_relation_set'

/var/www/dil_hydra/releases/20161212165014/app/models/multiresimage.rb:126:in `create_vra_work'

/var/www/dil_hydra/releases/20161212165014/app/models/multiresimage.rb:166:in `vra_save'
```

### Expected behavior

If a record fails due to error during batch ingest, all artifacts should be removed from the Images application.

### Actual behavior

If a record fails during some aspects of batch ingest, the VRAWork record is left in the Images application.

### Steps to reproduce the behavior

1. Perform a Batch Ingest where the xml file is missing the relationSet 
 ```
      <!--Relation-->
      <vra:relationSet>
         <vra:relation pref="true" type="imageOf" relids=""/>
      </vra:relationSet>
```

2. Observe that the record fails the first try with this error
```
2016-12-12T20:27:04.419Z 2250 TID-ow6wscu0g WARN: {"class"=>"MultiresimagesBatchWorker", "args"=>["/dropbox/batches/jy_afri_small/inu_afrmap_3797082_recto_master.tif"], "retry"=>true, "queue"=>"default", "jid"=>"b6c156b9e8f96f886ed6407a", "created_at"=>1481574422.9813397, "enqueued_at"=>1481574422.9813964, "error_message"=>"undefined method `set_attribute' for nil:NilClass", "error_class"=>"NoMethodError", "failed_at"=>1481574424.4194033, "retry_count"=>0}

2016-12-12T20:27:04.420Z 2250 TID-ow6wscu0g WARN: NoMethodError: undefined method `set_attribute' for nil:NilClass

2016-12-12T20:27:04.420Z 2250 TID-ow6wscu0g WARN: /var/www/dil_hydra/releases/20161212165014/app/models/vrawork.rb:67:in `update_relation_set'
/var/www/dil_hydra/releases/20161212165014/app/models/multiresimage.rb:126:in `create_vra_work'
/var/www/dil_hydra/releases/20161212165014/app/models/multiresimage.rb:166:in `vra_save'
```

3. Subsequent tries will fail with existing accesssion number
```
2016-12-12T20:27:53.811Z 2250 TID-ow6wtmuqo WARN: {"class"=>"MultiresimagesBatchWorker", "args"=>["/dropbox/batches/jy_afri_small/inu_afrmap_3797082_recto_master.tif"], "retry"=>true, "queue"=>"default", "jid"=>"b6c156b9e8f96f886ed6407a", "created_at"=>1481574422.9813397, "enqueued_at"=>1481574473.7996879, "error_message"=>"Existing image found with this accession number: inu_afrmap_3797082_recto_master", "error_class"=>"RuntimeError", "failed_at"=>1481574424.4194033, "retry_count"=>1, "retried_at"=>1481574473.8113728}

2016-12-12T20:27:53.811Z 2250 TID-ow6wtmuqo WARN: RuntimeError: Existing image found with this accession number: inu_afrmap_3797082_recto_master

2016-12-12T20:27:53.811Z 2250 TID-ow6wtmuqo WARN: /var/www/dil_hydra/releases/20161212165014/app/workers/multiresimages_batch_worker.rb:20:in `perform'
```

### Related work

#345

**Comment by davidschober** [comment]
@jenyoung can you review and OK to merge/close?

**Comment by jenyoung** [comment]
do you want me to intentionally send a duplicate record through batch ingest?  i'm not sure how to test since i don't see the backend.

**Comment by davidschober** [comment]
@kdid just putting in a batch that purposefully fails should do the trick, correct/

**Comment by kdid** [comment]
Above, under "Steps to Reproduce...." there were certain conditions that caused the orphaned work record. Like not including the relation set. 

Not all types of failures caused this. 

So not including the relation set will still fail. 

but subsequent retries wouldn't produce the duplicate accession number error....

**Comment by jenyoung** [comment]
@kdid I'm ready to test this now. I will send one record through without relationSet, and then send the same record through again with relationSet after it fails the first time.  Is this the accurate test?

**Comment by davidschober** [comment]
@kdid answered yes on slack.

**Comment by jenyoung** [comment]
Uploaded record without relationSet - failed. Uploaded same record with relationSet - passed and imported correctly. Image is giving the HTTP 500 attempting to load TileSource note but that might be ok. It's a large tiff.

**Comment by davidschober** [comment]
It has to crunch the image.

**Comment by jenyoung** [comment]
it's not giving the HTTP 500 error anymore, but it's not loading the image. In Sidekiq,  this error appears: "RuntimeError: Existing image found with this accession number: 2016_12_14_10_28_jen"

**Comment by davidschober** [comment]
OK. the @nulib/repodev team is cranking on getting IR out the door. Hang tight.

**Comment by kdid** [comment]
@jyo, I just tested the problem you were describing yesterday in images-s. I did a one record batch with no relationSet. I saw it fail in sidekiq with the `Had a problem saving /dropbox/batches/ksd_02162017/0216201701.tif: undefined method `set_attribute' for nil:NilClass` error. Then I killed thatjob in sidekiq under retries. Then I added the relationSet to the record and tried batch again and it worked. So it seems to be working fine.

the server was really slow and out of memory so I had devops restart it and it seems better for now.
